src/app/auth/account/account.component.ts

```diff
@@ -1,8 +1,190 @@
+                    filter((event) => event instanceof NavigationEnd),
+                    map((event) => event as RouterEvent),
+                )
+                .subscribe(() => this.getDatasets()),
+        );
+        if (this.user) {
+            this.accountName = this.user.login;
+            this.getDatasets();
+        }
+    }
+
+    public get isAccountViewTypeOverview(): boolean {
+        return this.accountViewType === AccountTabs.overview;
+    }
+
+    public get isAccountViewTypeDatasets(): boolean {
+        return this.accountViewType === AccountTabs.datasets;
+    }
+
+    public get isAccountViewTypeOrganizations(): boolean {
+        return this.accountViewType === AccountTabs.organizations;
+    }
+
+    public get isAccountViewTypeInbox(): boolean {
+        return this.accountViewType === AccountTabs.inbox;
+    }
+
+    public get isAccountViewTypeStars(): boolean {
+        return this.accountViewType === AccountTabs.stars;
+    }
+
+    public onEditProfile(): void {
+        promiseWithCatch(
+            this.modalService.warning({
+                message: AppValues.UNIMPLEMENTED_MESSAGE,
+                yesButtonText: "Ok",
+            }),
+        );
+    }
+
+    public selectedTabs(accountTabKey: AccountTabs): boolean {
+        if (this._window.location.search.includes(accountTabKey)) {
+            return true;
+        }
+        return false;
+    }
+
+    public onSelectOverviewTab(): void {
+        this.navigateTo(AccountTabs.overview);
+    }
+    public onSelectDatasetsTab(): void {
+        this.navigateTo(AccountTabs.datasets);
+    }
+
+    public onSelectOrganizationsTab(): void {
+        this.navigateTo(AccountTabs.organizations);
+    }
+
+    public onSelectInboxTab(): void {
+        this.navigateTo(AccountTabs.inbox);
+    }
+
+    public onSelectStarsTab(): void {
+        this.navigateTo(AccountTabs.stars);
+    }
+
+    public onSelectDataset(row: DatasetSearchOverviewFragment): void {
+        this.navigationService.navigateToDatasetView({
+            accountName: (row.owner as User).name,
+            datasetName: row.name as string,
+        });
+    }
+
+    public onPageChange(params: {
+        currentPage: number;
+        isClick: boolean;
+    }): void {
+        params.currentPage
+            ? (this.currentPage = params.currentPage)
+            : (this.currentPage = 1);
+        if (this.currentPage === 1) {
+            this.navigationService.navigateToOwnerView(
+                this.accountName,
+                this.accountViewType,
+            );
+            return;
+        }
+        this.navigationService.navigateToOwnerView(
+            this.accountName,
+            this.accountViewType,
+            params.currentPage,
+        );
+    }
+
+    private navigateTo(type: AccountTabs): void {
+        this.navigationService.navigateToOwnerView(this.accountName, type);
+        this.accountViewType = type;
+    }
+
+    private getDatasets(): void {
```

Comment:
Please create a little service, components should not be interacting with GraphQL directly.